src/seedu/addressbook/AddressBook.java

```diff
@@ -568,6 +600,39 @@ private static String executeClearAddressBook() {
         return MESSAGE_ADDRESSBOOK_CLEARED;
     }
 
+    /**
+     * Updates person identified using last displayed index.
+     *
+     * @param commandArgs full command args string from the user
+     * @return feedback display message for the operation result
+     */
+    private static String executeUpdatePerson(String commandArgs)   {
+        final String[] updateStringSplit = splitIndexAndPersonData(commandArgs);
+
+        if (!isDeletePersonArgsValid(updateStringSplit[0])) {
+            return getMessageForInvalidCommandInput(COMMAND_UPDATE_WORD, getUsageInfoForUpdateCommand());
+        }
+        final int targetVisibleIndex = extractTargetIndexFromDeletePersonArgs(updateStringSplit[0]);
+        if (!isDisplayIndexValidForLastPersonListingView(targetVisibleIndex)) {
+            return MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
+        }
+        final String[] targetInModel = getPersonByLastVisibleIndex(targetVisibleIndex);
+        if (deletePersonFromAddressBook(targetInModel))   {
+            // try decoding a person from the raw args
+            final Optional<String[]> decodeResult = decodePersonFromString(updateStringSplit[1]);
+
+            // checks if args are valid (decode result will not be present if the person is invalid)
+            if (!decodeResult.isPresent()) {
+            return getMessageForInvalidCommandInput(COMMAND_UPDATE_WORD, getUsageInfoForUpdateCommand());
+            }
+            // add the person as specified
+            final String[] personToUpdate = decodeResult.get();
+            addPersonToAddressBook(personToUpdate);
+            return getMessageForSuccessfulUpdatePerson(personToUpdate);// success
+            }   else return MESSAGE_PERSON_NOT_IN_ADDRESSBOOK; // not found
```

Comment:
Violation of code standard! Missing braces for the else branch. In addition, your indentation seems to be off.